SpatialGDK/Source/SpatialGDKServices/Public/SpatialPackageManager.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Async/Future.h"
+#include "CoreMinimal.h"
+#include "Misc/MonitoredProcess.h"
+#include "TimerManager.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialPackageManager, Log, All);
+
+class SPATIALGDKSERVICES_API FSpatialPackageManager
+{
+public:
+	FSpatialPackageManager();
+
+	static void FetchRuntimeBinary(FString RuntimeVersion);
+	static void FetchInspectorBinary(FString InspectorVersion);
+	void StartProcess(FString Params, FString ProcessName);
+	bool KillProcess(FString ProcessName);
+
+private:
+	TOptional<FMonitoredProcess> FetchingProcess = {};
```

Comment:
Does this need to be an optional?

Reply:
We are not entirely sure :D I think it has something to do with the async timeout; we tried to change it but couldn't get it to work without TOptional

Reply:
I reckon we can make this a unique pointer. FSpatialGDKPackageAssembly has an example using a shared pointer.

Reply:
hmmmm I think we tried with TSharedPtr and wasn't successful

Reply:
Using this logic: MakeShareable(new FMonitoredProcess());

Reply:
think so, yep

Reply:
do you remember why it failed?